examples/logs-slack-alerts/main.py

```diff
@@ -0,0 +1,60 @@
+import logging
+import requests
+from google.cloud import bigquery
+
+
+logging.getLogger().setLevel(logging.INFO)
+
+#Fill in variables with your project setup
+variables ={
+  "SLACK_WEBHOOK_URL": "https://hooks.slack.com/services/T00000000/B00000000/XXXXXXXXXXX",
+  "DATASET_NAME": "mydataset",
+  "AUDIT_LOG_TABLE":  "audit_table_name",
+  "TIME_COLUMN": "time_column",
+  "ERROR_MESSAGE_COLUMN":"error_message_column"
+}
+QUERY = """
+WITH
+  errors AS (
+  SELECT
+    {ERROR_MESSAGE_COLUMN} AS error_message,
+    EXTRACT(HOUR FROM current_timestamp) as hr,
+  FROM
+    {DATASET_NAME}.{AUDIT_LOG_TABLE}
+  WHERE
+    {ERROR_MESSAGE_COLUMN} IS NOT NULL
+    AND EXTRACT(HOUR
+    FROM
+      current_timestamp) = EXTRACT(HOUR
+    FROM
+     {TIME_COLUMN}))
+SELECT
+  error_message as Error,
+  hr,
+  COUNT(*) as Count
+FROM
+  errors
+GROUP BY
+  1,2
+""".format(**variables)
+
+def query_for_errors(incoming_request):
+
+  bq_client = bigquery.Client()
```

Comment:
Consider making this global that way its not recreated on every function invocation: https://cloud.google.com/functions/docs/bestpractices/tips#use_global_variables_to_reuse_objects_in_future_invocations

Reply:
Done